bundles/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/discovery/BigAssFanDiscoveryService.java

```diff
@@ -242,22 +248,28 @@ private synchronized void deviceDiscovered(BigAssFanDevice device) {
                 .withRepresentationProperty(THING_PROPERTY_MAC).withLabel(device.getLabel()).build());
     }
 
-    private void schedulePollJob() {
-        logger.debug("Scheduling discovery poll job to run every {} seconds starting in {} sec", POLL_FREQ, POLL_DELAY);
-        cancelPollJob();
-        pollJob = scheduler.scheduleWithFixedDelay(() -> {
-            try {
-                discoveryListener.pollForDevices();
-            } catch (RuntimeException e) {
-                logger.warn("Poll job got unexpected exception: {}", e.getMessage(), e);
-            }
-        }, POLL_DELAY, POLL_FREQ, TimeUnit.SECONDS);
+    private synchronized void schedulePollJob() {
+        if (this.pollJob == null) {
```

Comment:
Probably you want local variable here, see other comments.

Reply:
https://github.com/openhab/openhab-addons/pull/13903#discussion_r1057289352